src/gateways/ElasticsearchGateway.ts

```diff
@@ -63,17 +63,19 @@ export class ElasticsearchGateway {
       .log('[elasticsearch] searching documents');
 
     const conditionsArray = Object.entries(metadata).map(([key, value]) => ({
-      match: { [key]: value },
+      terms: { [key]: value.constructor === Array ? value : [value] },
```

Comment:
There is a built-in method for checking if something is an array that you can use, although this method is also pretty safe.

```suggestion
      terms: { [key]: Array.isArray(value) ? value : [value] },
```

Reply:
Sweet! Didn't realise it!